### PR TITLE
add: kwargs test and helper methods

### PR DESCRIPTION
**Pull request by harshdoesdev**
Adds Rust test for `kw-args` feature and a few helper methods for retrieving kwarg values.

### CODE REVIEW

File: ftd/src/interpreter/test.rs [hunk 1]
```diff
@@ -258,3 +258,41 @@ fn evalexpr_test() {
...
+    // TODO: resolve value based on kwarg kind
+    let data: std::collections::BTreeMap<String, String> = kwargs
+        .iter()
+        .map(|(name, property_value)| {
+            (
+                name.to_string(),
+                property_value.to_value().get_string_data().unwrap(),
+            )
+        })
+        .collect();
```
Comment (amitu) [inline ftd/src/interpreter/test.rs:294]: Can you make this two lines?

Comment (harshdoesdev) [reply ftd/src/interpreter/test.rs:294]: the first few lines are boilerplate code. Do you mean the `let data = ..` line?

Comment (amitu) [reply ftd/src/interpreter/test.rs:294]: I mean can you take lines I have highlighted and make it two lines. Remove the boilerplace code.

File: ftd/src/interpreter/test.rs [hunk 2]
```diff
@@ -258,3 +258,41 @@ fn evalexpr_test() {
...
+    -- ftd.text: Hello world
+
+    -- end: foo
+
+    -- foo:
+    bar: Hello
+    baz: World
+"#,
+    )
+    .unwrap_or_else(|e| panic!("{:?}", e));
```
Comment (amitu) [inline ftd/src/interpreter/test.rs:278]: This is also wrong. What is the code end user is supposed to write, write that code. Is interpret_helper in your opinion the method we should expose?

Comment (amitu) [reply ftd/src/interpreter/test.rs:278]: Point is not to test the "public API" for getting data. You have built a feature to extract data, just show me how am I supposed to use it.

File: ftd/src/interpreter/test.rs [hunk 3]
```diff
@@ -258,3 +258,31 @@ fn evalexpr_test() {
...
+    -- foo:
+    bar: Hello
+    baz: World
+"#,
+    )
+    .unwrap();
+
+    let tdoc = doc.tdoc();
+    let instructions = doc.get_instructions("foo#foo");
+    let instruction = instructions.first().unwrap();
```
Comment (amitu) [inline ftd/src/interpreter/test.rs:282]: This is still not good. tdoc is temporary doc, and is internal property. Can you write a new crate, that uses ftd crate a dependency, and see what code you will write. Test against that code.

File: ftd/src/interpreter/test.rs [hunk 4]
```diff
@@ -261,27 +191,26 @@ fn evalexpr_test() {
...
+        "#,
     )
     .unwrap();
 
-    let tdoc = doc.tdoc();
-    let instructions = doc.get_instructions("foo#foo");
-    let instruction = instructions.first().unwrap();
-
-    let data = instruction.get_kwargs("data", &tdoc).unwrap();
+    let component = doc.get_component_by_id("foo#fizz", "test").unwrap();
```
Comment (amitu) [inline ftd/src/interpreter/test.rs:212]: I would prefer if first argument was not needed. It should act like dom.get_element_by_id() in HTML (its playing exactly the same role, so why not copy their obviously superior behaviour).

**Review by amitu — APPROVED** [review]
Good job!

**Pull request merged**